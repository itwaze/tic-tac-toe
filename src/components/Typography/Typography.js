import React from "react";
import styled from "styled-components";

const StyledSpan = styled.span`
	font-size: ${({ variant }) => (variant === "body1" ? "24px" : "16px")};
	margin: ${({ margin }) => margin};
	display: block;
	width: fit-content;
`;

const Typography = (props) => {
	const { children, ...rest } = props;
	return <StyledSpan {...rest}>{children}</StyledSpan>;
};

export default Typography;
