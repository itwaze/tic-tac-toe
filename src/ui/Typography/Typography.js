import React from "react";
import styled from "styled-components";

const StyledSpan = styled.span`
  margin: ${({ margin }) => margin};
  font-size: ${({ fontSize }) => fontSize};
  display: block;
  width: fit-content;
`;

const Typography = (props) => {
  const { children, ...rest } = props;
  return <StyledSpan {...rest}>{children}</StyledSpan>;
};

export default Typography;
