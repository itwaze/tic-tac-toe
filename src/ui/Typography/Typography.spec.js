import React from "react";
import { render } from "@testing-library/react";

import Typography from "./Typography";

describe("Typography", () => {
  it("should take a snapshot", () => {
    const { asFragment } = render(<Typography>test text</Typography>);
    expect(asFragment()).toMatchSnapshot();
  });
});
