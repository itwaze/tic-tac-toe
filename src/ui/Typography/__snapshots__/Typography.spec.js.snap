// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Typography should take a snapshot 1`] = `
<DocumentFragment>
  <span
    class="sc-bdfBwQ gDKUml"
  >
    test text
  </span>
</DocumentFragment>
`;
